swift/ArrowFlight/Tests/ArrowFlightTests/FlightTest.swift

```diff
@@ -0,0 +1,291 @@
+        for try await rb in reader {
+            XCTAssertEqual(rb.schema.fields.count, 3)
+            XCTAssertEqual(rb.length, 4)
+            try await writer.write(FlightPutResult())
+        }
+    }
+    
+    func doGet(_ ticket: ArrowFlight.FlightTicket, writer: ArrowFlight.RecordBatchStreamWriter) async throws {
+        try await writer.write(try makeRecordBatch())
+    }
+    
+    func getSchema(_ request: ArrowFlight.FlightDescriptor) async throws -> ArrowFlight.FlightSchemaResult {
+        XCTAssertEqual(String(bytes: request.cmd, encoding: .utf8)!, "schema info")
+        XCTAssertEqual(request.type, .cmd)
+        return try ArrowFlight.FlightSchemaResult(schemaToArrowStream(makeSchema()))
+    }
+    
+    func getFlightInfo(_ request: ArrowFlight.FlightDescriptor) async throws -> ArrowFlight.FlightInfo {
+        return ArrowFlight.FlightInfo(Data())
+    }
+    
+    func listFlights(_ criteria: ArrowFlight.FlightCriteria, writer: ArrowFlight.FlightInfoStreamWriter) async throws {
+        XCTAssertEqual(String(bytes: criteria.expression, encoding: .utf8), "flight criteria expression")
+        let flight_info = try ArrowFlight.FlightInfo(schemaToArrowStream(makeSchema()))
+        try await writer.write(flight_info)
+    }
+    
+    func listActions(_ writer: ArrowFlight.ActionTypeStreamWriter) async throws {
+        try await writer.write(FlightActionType("type1", description: "desc1"))
+        try await writer.write(FlightActionType("type2", description: "desc2"))
+    }
+    
+    func doAction(_ action: FlightAction, writer: ResultStreamWriter) async throws {
+        XCTAssertEqual(action.type, "test_action")
+        XCTAssertEqual(String(bytes: action.body, encoding: .utf8)!, "test_action body")
+        try await writer.write(FlightResult("test_action result".data(using: .utf8)!))
+    }
+}
+
+struct FlightServerImpl {
+    var port = 1234
+    static var server: Server?
+    static var group: MultiThreadedEventLoopGroup?
+    static func run() async throws {
+        do {
+            // Create an event loop group for the server to run on.
+            let group = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
+            // Create a provider using the features we read.
+            let provider = ArrowFlight.MakeFlightServer(MyFlightServer())
+            
+            // Start the server and print its address once it has started.
+            FlightServerImpl.server = try await Server.insecure(group: group)
+                .withServiceProviders([provider])
+                .bind(host: "localhost", port: 8088)
+                .get()
+            
+            print("server started on port \(server!.channel.localAddress!.port!)")
+
+            // Wait on the server's `onClose` future to stop the program from exiting.
+        }catch {
+            print("Unknown server error: \(error)")
+        }
+    }
+}
+
+public class FlightClientImpl {
```

Comment:
`FlightClientTester` or something may be better.

Reply:
Will change to FlightClientTester.